### ISSUE TITLE
Custom skips can be hidden after already showing the question

### ISSUE THREAD

**Issue by wyattis** [human-nature-lab/trellis-app#445]
This logic fails because its evaluation changes just before arriving at the page that should be skipped.
```javascript
function showIf({ data, getEdge, followUpDatum }) {
  function countSetDifferences (newSet, oldSet) {
    let numDifferences = 0
    for (const id of newSet) {
      if (!oldSet.has(id)) {
        numDifferences++
      }
    }
    for (const id of oldSet) {
      if (!newSet.has(id)) {
        numDifferences++
      }
    }
    return numDifferences
  }
  
  const followUpEdge = getEdge(followUpDatum.edgeId)
  if (!followUpEdge) {
    return true
  }
  
  // Should be hidden if there are more than 10 changes in the friend network
  const { a1500, a1600, a2100, a2705, a2706, a2707, ab9990: previous_friends } = data
  const allFriends = a1500.allEdges.concat(a1600.allEdges).concat(a2100.allEdges).concat(a2705.allEdges).concat(a2706.allEdges).concat(a2707.allEdges)
  const newFriendsSet = new Set(allFriends.filter(e => !!e).map(e => e.targetRespondentId))
  const oldFriendsSet = new Set(previous_friends.allEdges.filter(e => !!e).map(e => e.targetRespondentId))
  const nDiff = countSetDifferences(newFriendsSet, oldFriendsSet)
  if (nDiff > 10) {
    return false
  }

  // Should show if a1500, a1600 and a2100 don't have this member
  const wasMentioned = newFriendsSet.has(followUpEdge.targetRespondentId)
  return !wasMentioned
}
```

**Comment by wyattis** [comment]
Fixed by 0e1c635d7e86046bd2c20136af01518cd21b6db8

**Comment by wyattis** [comment]
Not fully fixed

**Comment by wyattis** [comment]
Fixed in 53d3629d6603403d1a38af8062ea996c7c80ea85
